ios/FluentUI/Navigation/BarButtonItemView.swift

```diff
@@ -0,0 +1,271 @@
+        badgeView.textColor = .white
+        badgeView.textAlignment = .center
+        badgeView.font = UIFont.systemFont(ofSize: Constants.badgeFontSize, weight: .regular)
+        badgeView.isHidden = true
+
+        return badgeView
+    }()
+
+    private func updateLayout() {
+        updateButton()
+        updateBadgeView()
+        invalidateIntrinsicContentSize()
+    }
+
+    private func updateButton() {
+        button.isEnabled = item.isEnabled
+
+        if isLeftItem {
+            let isRTL = effectiveUserInterfaceLayoutDirection == .rightToLeft
+            button.contentEdgeInsets = UIEdgeInsets(top: 0, left: isRTL ? 0 : Constants.leftBarButtonItemLeadingMargin, bottom: 0, right: isRTL ? Constants.leftBarButtonItemLeadingMargin : 0)
+        } else {
+            button.contentEdgeInsets = UIEdgeInsets(top: 0, left: Constants.rightBarButtonItemHorizontalPadding, bottom: 0, right: Constants.rightBarButtonItemHorizontalPadding)
+        }
+
+        button.tag = item.tag
+        button.tintColor = item.tintColor
+        button.titleLabel?.font = item.titleTextAttributes(for: .normal)?[.font] as? UIFont
+
+        var portraitImage = item.image
+        if portraitImage?.renderingMode == .automatic {
+            portraitImage = portraitImage?.withRenderingMode(.alwaysTemplate)
+        }
+        var landscapeImage = item.landscapeImagePhone ?? portraitImage
+        if landscapeImage?.renderingMode == .automatic {
+            landscapeImage = landscapeImage?.withRenderingMode(.alwaysTemplate)
+        }
+
+        button.setImage(traitCollection.verticalSizeClass == .regular ? portraitImage : landscapeImage, for: .normal)
+        button.setTitle(item.title, for: .normal)
+
+        if let action = item.action {
+            button.addTarget(item.target, action: action, for: .touchUpInside)
+        }
+
+        button.showsLargeContentViewer = true
+
+        if let customLargeContentSizeImage = item.largeContentSizeImage {
+            button.largeContentImage = customLargeContentSizeImage
+        }
+
+        if item.title == nil {
+            button.largeContentTitle = item.accessibilityLabel
+        }
+
+        if #available(iOS 13.4, *) {
+            // Workaround check for beta iOS versions missing the Pointer Interactions API
+            if arePointerInteractionAPIsAvailable() {
+                button.isPointerInteractionEnabled = true
+            }
+        }
+    }
+
+    private func updateBadgeView() {
+        badgeView.text = badgeValue
+        badgeView.isHidden = badgeValue == nil
+
+        if badgeValue != nil {
+            let maskLayer = CAShapeLayer()
+            maskLayer.fillRule = .evenOdd
+
+            let path = UIBezierPath(rect: CGRect(x: 0, y: 0, width: button.frame.size.width, height: button.frame.size.height))
+            let badgeVerticalOffset = Constants.badgeVerticalOffset
+
+            if badgeView.text?.count ?? 1 > 1 {
```

Comment:
<img width="336" alt="Screenshot 2021-09-17 at 2 10 33 PM" src="https://user-images.githubusercontent.com/33866787/133752672-3fd1cab3-f521-4e37-b7ff-332e48e189bc.png">

We have exactly the same thing in the case of side tab bar items also. It happens with more than 3 characters.

Reply:
It seems like in the Navigation Bar case it's allowed 3 characters in addition to the ellipsis ("...") which makes the text a little longer.
If you compare a close screenshot of the 2 cases above in this comment there is a difference in how close the last dot is from the edge of the badge:

| Side Bar | Navigation Bar |
| --------------- | --------------- |
| ![Screen Shot 2021-09-17 at 3 29 05 PM](https://user-images.githubusercontent.com/68076145/133860276-70a102ef-7e33-4a9f-ba18-dda35c57cf23.png) | ![Screen Shot 2021-09-17 at 3 28 50 PM](https://user-images.githubusercontent.com/68076145/133860274-c20b30c3-ddcc-44f6-90e8-0c59980ed874.png) |

Reply:
Fixed!! Also, the max width for the badge coming on side tab bar items needs to be different then the width of the badge view coming on bar button items and the reason is that we have more space in case of side tab bar or bottom tab bar items as compared to the bar button items in nav bar.